### PR TITLE
ci: update x86_64-macos-release to macos-13

### PR DESCRIPTION
**Pull request by nektro**
Closes https://github.com/ziglang/zig/issues/20467

### CODE REVIEW

**Comment by nektro** [comment]
runner acquisition, initial boostrap, stage2 behavior tests ran successfully. llvm-enabled build failed init

```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CURSES
    linked by target "zigcpp" in directory /Users/runner/work/zig/zig
```

**Comment by mikdusan** [comment]
Because `-DZIG_STATIC=ON` it is eventually expecting static curses library. Perhaps try:
```diff
diff --git a/ci/x86_64-macos-release.sh b/ci/x86_64-macos-release.sh
index acf2b91d9..9f8671b58 100755
--- a/ci/x86_64-macos-release.sh
+++ b/ci/x86_64-macos-release.sh
@@ -50,6 +50,7 @@ cmake .. \
   -DZIG_TARGET_TRIPLE="$TARGET" \
   -DZIG_TARGET_MCPU="$MCPU" \
   -DZIG_STATIC=ON \
+  -DZIG_STATIC_CURSES=OFF \
   -DZIG_NO_LIB=ON
 
 make $JOBS install
```

hopefully stage4 `zig build ...` is able to pickup on this choice

**Comment by nektro** [comment]
is `-DZIG_STATIC=ON -DZIG_STATIC_CURSES=OFF` not contradictory? makes me think we're pulling in curses unintentionally or we need to install it explicitly so that the static library is available?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
       - name: Build and Test
         run: sh ci/aarch64-linux-release.sh
   x86_64-macos-release:
-    runs-on: "macos-11"
+    runs-on: "macos-13"
```
Comment (andrewrk) [inline .github/workflows/ci.yaml:46]:
```suggestion
    runs-on: "macos-12"
```

the CI makes sure things work on the oldest supported system

**Review by andrewrk — CHANGES_REQUESTED** [review]

**Comment by nektro** [comment]
superseded by https://github.com/ziglang/zig/pull/20473

**Pull request closed without merging**